http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -153,6 +155,9 @@ private HttpRequestBase wrapEntity(HttpExecuteRequest request,
             if (!request.httpRequest().firstMatchingHeader(HttpHeaders.CONTENT_LENGTH).isPresent()) {
                 entity = ApacheUtils.newBufferedHttpEntity(entity);
             }
+            if (request.httpRequest().matchingHeaders("Transfer-Encoding").contains("chunked")) {
```

Comment:
Can we use `Header.TRANSFER_ENCODING` and `Header.CHUNKED`? Also, let's use equals instead of contains

Reply:
updated to constant headers.
used matchingHeaders() which returns a List<String> , in the unlikely case there are multiple values in TE, which is why I used contains. What do you think?